File: tests/fixtures/sample.rb
# Sample Ruby file for testing tree-sitter parsing.

require 'json'
require_relative './helper'

module Greetable
  def greet(name)
    raise NotImplementedError
  end
end

class HelloGreeter
  include Greetable

  def initialize(prefix = 'Hello')
    @prefix = prefix
  end

  def greet(name)
    "#{@prefix}, #{name}!"
  end

  private

  def helper
    puts 'helper'
  end
end

class DataProcessor
  def process_data(data)
    sum = 0
    data.each do |v|
      if v > 0
        (0...v).each do |i|
          sum += i
        end
      end
    end
    sum
  end
end

def standalone_function(x)
  x * 2
end

if __FILE__ == $PROGRAM_NAME
  g = HelloGreeter.new
  puts g.greet('World')
end

File: tests/fixtures/sample.c
/* Sample C file for testing tree-sitter parsing. */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

/* Struct for greeter */
struct Greeter {
    char prefix[64];
};

/* Initialize greeter */
void greeter_init(struct Greeter *g, const char *prefix) {
    strncpy(g->prefix, prefix, sizeof(g->prefix) - 1);
    g->prefix[sizeof(g->prefix) - 1] = '\0';
}

/* Greet function */
void greeter_greet(struct Greeter *g, const char *name) {
    printf("%s, %s!\n", g->prefix, name);
}

/* Process data with nested loops */
int process_data(int *data, int len) {
    int sum = 0;
    for (int i = 0; i < len; i++) {
        if (data[i] > 0) {
            for (int j = 0; j < data[i]; j++) {
                sum += j;
            }
        }
    }
    return sum;
}

/* Helper function */
static int helper(int x) {
    return x * 2;
}

int main(int argc, char *argv[]) {
    struct Greeter g;
    greeter_init(&g, "Hello");
    greeter_greet(&g, "World");
    return 0;
}
